Lib/asyncio/sslproto.py

```diff
@@ -571,21 +571,17 @@ def _on_handshake_complete(self, handshake_exc):
             self._handshake_timeout_handle = None
 
         sslobj = self._sslobj
-        try:
-            if handshake_exc is None:
-                self._set_state(SSLProtocolState.WRAPPED)
-            else:
-                raise handshake_exc
-
+        if handshake_exc is None:
+            self._set_state(SSLProtocolState.WRAPPED)
             peercert = sslobj.getpeercert()
-        except Exception as exc:
```

Comment:
The exception handler here is intended to (also) catch exceptions raised by `sslobj.getpeercert()`. That is probably going to be an issue.

Reply:
In other words: you can't avoid using a `try...except` block here. Is removing this really necessary to break the cycle?

Reply:
Yup! This was my misunderstanding of the problem. It turns out that setting the original `handshake_exc` to None also fixed our problem -- which my understanding is that there are two separate instances of the exception, each with a traceback and only one was being handled.

```py
try:
    if handshake_exc is None:
        self._set_state(SSLProtocolState.WRAPPED)
    else:
        raise handshake_exc

    peercert = sslobj.getpeercert()
except Exception as exc:
    handshake_exc = None    # <--- fixes the problem
```